src/main/java/foundation/privacybydesign/email/EmailRestApi.java

```diff
@@ -58,8 +66,24 @@ public Response sendEmail(@FormParam("email") String email,
             return Response.status(Response.Status.BAD_REQUEST).entity(ERR_ADDRESS_MALFORMED).build();
         }
 
-        String token = signer.createToken(email);
         try {
+            String ip = req.getHeader(PROXY_IP_HEADER);
+            if (ip == null) {
+                ip = req.getRemoteAddr();
+            }
```

Comment:
IP rate limiting is a bit outdated since we have IPv6. Did you consider only rate limiting on email address?

Reply:
Decided to remove IP rate limiting after further discussion since it does not bring any advantage while it could cause issues with shared IP environments.